### PR TITLE
testeth errors when an odd-length hex string encodes a byte sequence

### PR DESCRIPTION
**Pull request by pirapira**
For `code` and `data`, `testeth` now expects an even-length hex strings.

Whenever a sequence of bytes is encoded as hex, the result's length is even.

### CODE REVIEW

File: test/tools/libtesteth/TestHelper.cpp [hunk 1]
```diff
@@ -249,18 +253,25 @@ string compileLLL(string const& _code)
...
 	result = "0x" + result;
+	checkHexHasEvenLength(result);
 	return result;
 #endif
 }
 
+void checkHexHasEvenLength(string const& _str)
+{
+	if (_str.size() % 2)
+		BOOST_ERROR("An odd-length hex string represents a byte sequence: " + _str);
```
Comment (winsvega) [inline test/tools/libtesteth/TestHelper.cpp]: you should use TestOutputHelper::  to output current test case and test name

File: test/tools/libtesteth/TestHelper.cpp [hunk 2]
```diff
@@ -249,18 +253,25 @@ string compileLLL(string const& _code)
...
+
 bytes importCode(json_spirit::mObject& _o)
 {
 	bytes code;
 	if (_o["code"].type() == json_spirit::str_type)
 		if (_o["code"].get_str().find("0x") != 0)
 			code = fromHex(compileLLL(_o["code"].get_str()));
 		else
-			code = fromHex(_o["code"].get_str().substr(2));
+			code = importByteArray(_o["code"].get_str());
```
Comment (winsvega) [inline test/tools/libtesteth/TestHelper.cpp]: what about the case above? it still uses fromHex

**Comment by winsvega** [comment]
so there was no error after refilling corrected pairing test ?

File: test/tools/libtesteth/TestHelper.cpp (hunk 2 above)
Comment (pirapira) [inline test/tools/libtesteth/TestHelper.cpp]: In the above case, `compileLLL` checks the length of the returned string.

**Comment by pirapira** [comment]
@winsvega there were some other errors, but I fixed all of them in https://github.com/ethereum/tests/pull/190

**Pull request merged**